dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java

```diff
@@ -273,6 +273,34 @@ private Object getClassFromCDI(Class<?> clazz) {
         return serviceObject;
     }
 
+    /**
+     * @param clazz
+     * @return
+     */
+    @FFDCIgnore(value = { Exception.class })
+    private ManagedObject<?> getClassFromServiceObject(Class<?> clazz, Object serviceObject) {
+
+        if (! clazz.equals(serviceObject.getClass())) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Couldn't create object instance from ManagedObjectFactory for : " + clazz.getName() + "because the serviceObject and class had different types");
+            }
+            return null;
+        }
+
+        ManagedObjectFactory<Object> managedObjectFactory = (ManagedObjectFactory<Object>) getManagedObjectFactory(clazz);
+
+        ManagedObject<Object> bean = null;
+        try {
+            bean = managedObjectFactory.createManagedObject(serviceObject, null);
+        } catch (Exception e) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Couldn't create object instance from ManagedObjectFactory for : " + clazz.getName() + ", but ignore the FFDC: ", e);
```

Comment:
I'm thinking you might need warning or error messages here (and above) rather than just Tr.debugs

Reply:
I'm fine with leaving this as a debug - it is the same as what we currently have in `getClassFromManagedObject`.